components.tsx

```diff
@@ -77,36 +94,45 @@ const styles = StyleSheet.create({
   bottomNavigation: {
     flexDirection: "row",
     alignItems: "center",
-    marginBottom: 24
+    marginBottom: 32
   }
 });
 
-const PrimaryButton = (props: {
+type FontWeightVariants = "SemiBold" | "Medium" | "Bold";
+
+const AppText = (props: {
+  children: React.ReactNode;
+  style?: TextStyle | TextStyle[];
+  fontWeight?: FontWeightVariants;
+}) => {
+  const fontType = !props.fontWeight ? "Regular" : props.fontWeight;
+  return (
+    <Text style={[{ fontFamily: `Montserrat-${fontType}` }, props.style]}>
+      {props.children}
+    </Text>
+  );
+};
+
+const Button = (props: {
   text: string;
-  icon?: React.ReactNode;
   onPress: () => void;
+  secondary?: boolean;
```

Comment:
I'm always a bit concerned regarding boolean flags like this, what happens when you need a third kind of button?
Material UI uses a `variant` prop with is typed like your fontWeight in AppText, alternatively separate components like "Button", "SecondaryButton" with a uniform prop interface.

Reply:
good point. changed it to the `variant` suggestion.